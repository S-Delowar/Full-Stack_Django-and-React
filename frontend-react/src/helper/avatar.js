 export const randomAvatar = () =>
 `https://i.pravatar.cc/300?img=${Math.floor(Math.
 random() * 60) + 1}`;

export function getAvatarURL(avatarPath) {
  const BASE_URL = "http://localhost:8000";
  return avatarPath ? `${BASE_URL}${avatarPath}` : randomAvatar();
}