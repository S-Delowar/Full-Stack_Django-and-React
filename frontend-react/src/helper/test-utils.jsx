// test-utils.jsx
import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { Context } from "../components/Layout"; // âœ… your context import

function render(ui, { providerProps = {}, ...renderOptions } = {}) {
  const Wrapper = ({ children }) => (
    <BrowserRouter>
      <Context.Provider value={providerProps}>
        {children}
      </Context.Provider>
    </BrowserRouter>
  );

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}


export * from "@testing-library/react";
export { render };
