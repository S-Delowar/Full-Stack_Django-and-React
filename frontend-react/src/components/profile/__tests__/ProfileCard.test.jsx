import { render, screen } from "../../../helper/test-utils";
import TestRenderer from "react-test-renderer";
import ProfileCard from "../ProfileCard";
import { BrowserRouter } from "react-router-dom";
import { expect } from "vitest";

const userData = {
  id: "0590cd67-eacd-4299-8413-605bd547ea17",
  first_name: "Mossie",
  last_name: "Murphy",
  post_count: 3,
  email: "Mossie@yopmail.com",
  username: "MossieMurphy",
  avatar: null,
  created: "2022-08-19T17:31:03.310Z",
  updated: "2022-08-20T07:38:47.631Z",
};

test("Render ProfileCard component", () => {
  render(<ProfileCard profileUser={userData} />);

  const profileCard = screen.getByTestId("profile-card");
  expect(profileCard).toBeInTheDocument();

  const nameElement = screen.getByText(userData.username);
  expect(nameElement).toBeInTheDocument();
});

test("Profile Card snapshot", () => {
  const profileCardDomTree = TestRenderer.create(
    <BrowserRouter>
      <ProfileCard user={userData} />
    </BrowserRouter>
  ).toJSON();
  expect(profileCardDomTree).toMatchSnapshot();
});