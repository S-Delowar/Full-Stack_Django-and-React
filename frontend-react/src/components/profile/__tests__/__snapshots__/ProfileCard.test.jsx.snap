// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Profile Card snapshot 1`] = `null`;
